### PR TITLE
feat(extensions/crypto): implement verify() for RSA

### PR DESCRIPTION
**Pull request by littledivy**
Towards #1891

Introduces `SubtleCrypto.verify` for RSA algorithms (`RSASSA-PKCS1-v1_5` and `RSA-PSS`)

Originally the plan was to implement all current algorithms supported by `sign()` in this PR but we can't enable WPT for both the methods because it relies on `SubtleCrypto.importKey`.

### CODE REVIEW

**Comment by lucacasonato** [comment]
Make sure to update `supportedAlgorithms`.

File: extensions/crypto/00_crypto.js [hunk 1]
```diff
@@ -383,6 +387,113 @@
...
+     * @param {string} algorithm
+     * @param {CryptoKey} key
+     * @param {BufferSource} signature
+     * @param {BufferSource} data
+     * @returns {Promise<boolean>}
+     */
+    async verify(algorithm, key, signature, data) {
+      webidl.assertBranded(this, SubtleCrypto);
+      const prefix = "Failed to execute 'verify' on 'SubtleCrypto'";
+      webidl.requiredArguments(arguments.length, 3, { prefix });
```
Comment (lucacasonato) [inline extensions/crypto/00_crypto.js:400]:
```suggestion
      webidl.requiredArguments(arguments.length, 4, { prefix });
```

**Review by lucacasonato — APPROVED** [review]
JS side LGTM, except a minor comment.

@bnoordhuis could you review the Rust side crypto code?

**Review by bnoordhuis — APPROVED** [review]
LGTM.

The duplication for the SHA versions is kind of unfortunate. Maybe it's time to break out `macro_rules!`?

**Pull request merged**